pkg/sql/parser/builtins.go

```diff
@@ -1271,6 +1271,15 @@ var powImpls = []Builtin{
 		_, err := decimal.Pow(&dd.Dec, x, y, decimal.Precision)
 		return dd, err
 	}),
+	{
+		Types:      ArgTypes{TypeInt, TypeInt},
+		ReturnType: TypeInt,
+		fn: func(_ *EvalContext, args DTuple) (Datum, error) {
+			x := float64(*args[0].(*DInt))
+			y := float64(*args[1].(*DInt))
+			return NewDInt(DInt(math.Pow(x, y))), nil
```

Comment:
Also this is rather inefficient. If you know you're exponentiating an integer use exponentiation by squaring, see https://en.wikipedia.org/wiki/Exponentiation_by_squaring (use the iterative version, and simply return 0 if the exponent is negative)